x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java

```diff
@@ -60,20 +59,37 @@ protected void doStop() {}
     protected void doClose() {}
 
     @Override
-    public void onTaskRegistered(Task task) {
+    public void onRegistered(TracingPlugin.Traceable traceable) {
         final Tracer tracer = this.tracer;
         if (tracer != null) {
-            taskSpans.computeIfAbsent(task.getId(), taskId -> {
-                final Span span = tracer.spanBuilder(task.getAction()).startSpan();
-                span.setAttribute("es.task.id", task.getId());
+            spans.computeIfAbsent(traceable.getSpanId(), spanId -> {
+                final Span span = tracer.spanBuilder(traceable.getSpanName()).startSpan();
+                for (Map.Entry<String, Object> entry : traceable.getAttributes().entrySet()) {
+                    final Object value = entry.getValue();
+                    if (value instanceof String) {
+                        span.setAttribute(entry.getKey(), (String) value);
+                    } else if (value instanceof Long) {
+                        span.setAttribute(entry.getKey(), (Long) value);
+                    } else if (value instanceof Integer) {
+                        span.setAttribute(entry.getKey(), (Integer) value);
+                    } else if (value instanceof Double) {
+                        span.setAttribute(entry.getKey(), (Double) value);
+                    } else if (value instanceof Boolean) {
+                        span.setAttribute(entry.getKey(), (Boolean) value);
+                    } else {
+                        throw new IllegalArgumentException(
+                            "span attributes do not support value type of [" + value.getClass().getCanonicalName() + "]"
+                        );
+                    }
+                }
```

Comment:
This is not pretty. But it does the job for now and keep other parts unware of the opentelemetry dependencies. We can improve it once we know more about what other things needed by the new `Traceable` interface.